### PR TITLE
Updating template/composer.json to use Drupal packagist constraints.

### PR DESCRIPTION
**Pull request by grasmash**

### CODE REVIEW

**Comment by grasmash** [comment]
Todo:
- [ ]  Integrate script in update hook
- [ ]  Resolve drupal security advisory conflict (upstream in Lightning)

File: scripts/blt/composer-version-convert.php [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    return '*@stable';
+  }
+
+  // ~8.1.0-alpha1 > ~1.0.0-alpha1
+  // ^8.1.0-alpha1 > ^1.0.0-alpha1
+  // ^8.1.0 > ^1.0
+  // 8.1.0-alpha1 > 1.0.0-alpha1
+  // 8.1.0-beta12 > 1.0.0-beta12
+  // 8.12.0-rc22 > 12.0.0-rc22
+  if (preg_match('/^([\^~])?8(\.)?(\d+)?(\.\d+)?(-(alpha|beta|rc)\d+)?(\.\*)?(@(dev|alpha|beta|rc))?/', $version_constraint, $matches)) {
```
Comment (grasmash) [inline scripts/blt/composer-version-convert.php]: Need to also cover refs and "dev-" prefixes.

Comment (grasmash) [inline scripts/blt/composer-version-convert.php]: Need to support more operators:
https://getcomposer.org/doc/articles/versions.md#range

File: scripts/blt/composer-version-convert.php [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  'url' => 'https://packages.drupal.org/8',
+];
+
+$output = json_encode($composer_json, JSON_PRETTY_PRINT|JSON_UNESCAPED_SLASHES);
+print $output;
+
+function convert_version_constraint($version_constraint) {
+
+  // 8.* => *
+  if (preg_match('/8\.\*$/', $version_constraint)) {
```
Comment (phenaproxima) [inline scripts/blt/composer-version-convert.php]: There's no anchor at the beginning, so won't this also match something like 1.8.*?

**Comment by balsama** [comment]
We ended up tweaking the output for branches a little bit to:
* Account for the `-dev` coming at the end, e.g: `"drupal/google_analytics": "8.2.x-dev"`
* Ultimately convert all branches to `{#}.x-dev` as that seems to be the only documented format for branches on D.O packagist

See here: https://github.com/acquia/lightning/blob/8.x-1.x/src/UpgradeBuild.php

**Pull request merged**